File: 785-is-graph-bipartite/785-is-graph-bipartite.py
class Solution:
    def isBipartite(self, graph: List[List[int]]) -> bool:
        n = len(graph)
        color = [-1] * n
        color[0] = 0
        
        for i in range(n):
            for nbr in graph[i]:
                if color[nbr] == -1:
                    color[nbr] = 1 - color[i]
                
                elif color[i] == color[nbr]:
                    return False
        return True

File: 785-is-graph-bipartite/785-is-graph-bipartite.cpp
class Solution {
public:
    bool isBipartite(vector<vector<int>>& graph) {
        return solve(graph);
    }
private:
    bool solve(vector<vector<int>> &graph) {
        int n = graph.size();
        vector<int> color(n, -1);
        color[0] = 0;
        for(int i = 0; i < n ; i++) {
            for(auto x : graph[i]) {
                if(color[x] == -1)
                    color[x] = 1 - color[i];
                
                else if(color[x] == color[i])
                      return false;
            }
        }
        return true;
    }
};